### PR TITLE
[Snackbar] Implement a semantic color scheme color themer API.

### PR DESCRIPTION
**Pull request by yarneo**
Addition of a unit test, and updated the examples to use the new themer. Removed older stub color themer that was a no-op.

Pivotal: https://www.pivotaltracker.com/story/show/156171666

simulator screen shot - iphone 8 - 2018-04-08 at 14 08 29

### CODE REVIEW

File: components/Snackbar/src/ColorThemer/MDCSnackbarColorThemer.m
Comment (yarneo) [inline components/Snackbar/src/ColorThemer/MDCSnackbarColorThemer.m]: @jverkoey let me know if you find a better way to approach opacity for our theming.

File: components/Snackbar/src/ColorThemer/MDCSnackbarColorThemer.m
Comment (yarneo) [inline components/Snackbar/src/ColorThemer/MDCSnackbarColorThemer.m:25]: @jverkoey buttonTitleColor is still a question mark for me, I'm not sure of the right color for this so using existing color.

File: components/Snackbar/src/ColorThemer/MDCSnackbarColorThemer.h
Comment (jverkoey) [inline components/Snackbar/src/ColorThemer/MDCSnackbarColorThemer.h]: If there is no internal client usage, please mark this API as deprecated first. Once a release lands with this API having been marked deprecated then we can delete it.

File: components/Snackbar/src/ColorThemer/MDCSnackbarColorThemer.m
Comment (jverkoey) [reply components/Snackbar/src/ColorThemer/MDCSnackbarColorThemer.m]: For now this is the best way we have to approach opacity theming.

File: components/Snackbar/src/ColorThemer/MDCSnackbarColorThemer.m
Comment (jverkoey) [inline components/Snackbar/src/ColorThemer/MDCSnackbarColorThemer.m]: I'm surprised that the snackbar's background color isn't primary or secondary. "on surface" is typically reserved for text that will be drawn on top of the surface color. If "on surface" is what the params are expecting then it may be worth confirming that this is intentional.

File: components/Snackbar/src/ColorThemer/MDCSnackbarColorThemer.m
Comment (jverkoey) [reply components/Snackbar/src/ColorThemer/MDCSnackbarColorThemer.m:25]: Button title color should be using an "on" color, but because the background color is an "on" color this code gets a bit wonky...

**Review by jverkoey — CHANGES_REQUESTED** [review]

File: components/Snackbar/src/ColorThemer/MDCSnackbarColorThemer.m
Comment (yarneo) [reply components/Snackbar/src/ColorThemer/MDCSnackbarColorThemer.m]: Will confirm. I looked up the params and also commented there, but afaik snackbar is considered a "dark theme" component and hence uses the opposite theming; light content on dark surface rather than vice versa.

File: components/Snackbar/src/ColorThemer/MDCSnackbarColorThemer.h
Comment (yarneo) [reply components/Snackbar/src/ColorThemer/MDCSnackbarColorThemer.h]: are you sure we want to deprecate it? It was a newly added stub that was added in the last release and has no content. I feel like it would just be more redundant steps on our end.

Comment (jverkoey) [reply components/Snackbar/src/ColorThemer/MDCSnackbarColorThemer.h]: Deprecation as a practice helps external clients migrate to new versions of MDC. I agree that this API is unlikely to have been used, but it's safer for us in general if we always follow the deprecation policy for any public APIs.

Comment (yarneo) [reply components/Snackbar/src/ColorThemer/MDCSnackbarColorThemer.h]: done

**Review by jverkoey — APPROVED** [review]
Lgtm if this matches params, but please file a follow-up story to address the odd mappings (may require upstream design changes).

**Comment by yarneo** [comment]
Verification story: https://www.pivotaltracker.com/story/show/156652338

**Pull request merged**
